File: simulacrum/src/main/java/io/github/projectet/dmlSimulacrum/item/Items.java
package io.github.projectet.dmlSimulacrum.item;

import dev.nathanpb.dml.item.ItemsKt;
import io.github.projectet.dmlSimulacrum.dmlSimulacrum;
import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.item.Item;
import net.minecraft.item.ItemGroup;
import net.minecraft.util.registry.Registry;

public class Items {

    public static void Register() {
        Registry.register(Registry.ITEM, dmlSimulacrum.id("polymer_clay"), POLYMER_CLAY);
        Registry.register(Registry.ITEM, dmlSimulacrum.id("overworld_matter"), OVERWORLD_MATTER);
        Registry.register(Registry.ITEM, dmlSimulacrum.id("hellish_matter"), HELLISH_MATTER);
        Registry.register(Registry.ITEM, dmlSimulacrum.id("extraterrestrial_matter"), EXTRATERRESTRIAL_MATTER);
    }

    public static final Item POLYMER_CLAY = new ItemPolymerClay(new FabricItemSettings().maxCount(64).group(ItemsKt.getITEM_GROUP()));
    public static final Item OVERWORLD_MATTER = new ItemMatter(new FabricItemSettings().maxCount(64).group(ItemsKt.getITEM_GROUP()), dmlSimulacrum.config.Matter_XP.OverworldMatterXP);
    public static final Item HELLISH_MATTER = new ItemMatter(new FabricItemSettings().maxCount(64).group(ItemsKt.getITEM_GROUP()), dmlSimulacrum.config.Matter_XP.HellishMatterXP);
    public static final Item EXTRATERRESTRIAL_MATTER = new ItemMatter(new FabricItemSettings().maxCount(64).group(ItemsKt.getITEM_GROUP()), dmlSimulacrum.config.Matter_XP.ExtraMatterXP);
}

File: simulacrum/src/main/kotlin/dev/nathanpb/dml/simulacrum/item/Items.kt
package dev.nathanpb.dml.simulacrum.item

import dev.nathanpb.dml.config
import dev.nathanpb.dml.identifier
import dev.nathanpb.dml.itemgroup.ITEM_GROUP_KEY
import dev.nathanpb.dml.simulacrum.SIMULATION_CHAMBER
import net.fabricmc.fabric.api.item.v1.FabricItemSettings
import net.fabricmc.fabric.api.itemgroup.v1.ItemGroupEvents
import net.minecraft.item.ItemStack
import net.minecraft.registry.Registries
import net.minecraft.registry.Registry

val OVERWORLD_MATTER = ItemMatter(FabricItemSettings(), config.matterXP.overworldMatterXP)
val HELLISH_MATTER = ItemMatter(FabricItemSettings(), config.matterXP.hellishMatterXP)
val EXTRATERRESTRIAL_MATTER = ItemMatter(FabricItemSettings(), config.matterXP.extraterrestrialMatterXP)

fun registerItems() {
    linkedMapOf(
        EXTRATERRESTRIAL_MATTER to "extraterrestrial_matter",
        HELLISH_MATTER to "hellish_matter",
        OVERWORLD_MATTER to "overworld_matter"
    ).forEach { (item, id) ->
        Registry.register(Registries.ITEM, identifier(id), item)
        ItemGroupEvents.modifyEntriesEvent(ITEM_GROUP_KEY).register {
            it.addAfter(ItemStack(SIMULATION_CHAMBER), item)
        }
    }
}